src/couch/src/couch_bt_engine_header.erl

```diff
@@ -282,8 +282,13 @@ upgrade_epochs(#db_header{} = Header) ->
                 % compaction proceeds. In that case the epochs sequence could
                 % be regressing.
                 Epochs0;
-            [{_OtherNode, S} | _] = Epochs1 ->
-                assert_monotonic_update_seq(Header, S),
+            [{_OtherNode, S} | _] = Epochs0 when Header#db_header.update_seq < S ->
```

Comment:
Thanks for adding the clause here, this is much better than a try-catch block.

Reply:
Hmm, I thought this might be the safest option, given that we expect to handle this case in the remove_dup_epochs bellow. We run upgrade_header logic on open even if we don't commit commit and we've been handling that right below.